modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -84,24 +90,24 @@ func (p *Processor) Shutdown(_ context.Context) {
 
 func (p *Processor) aggregateMetrics(resourceSpans []*v1_trace.ResourceSpans) {
 	for _, rs := range resourceSpans {
-		// already extract service name, so we only have to do it once per batch of spans
-		svcName, _ := processor_util.FindServiceName(rs.Resource.Attributes)
-
+		// already extract job name & instance id, so we only have to do it once per batch of spans
+		jobName := processor_util.GetJobValue(rs.Resource.Attributes)
+		instanceID, _ := processor_util.FindInstanceID(rs.Resource.Attributes)
```

Comment:
I think we should handle the case when there is no job and instance ID. In that case, `target_info` can't be matched to the rest of the metrics.

Reply:
so right now if either of them are blank, it is set to "". What do you suggest should happen instead?

Reply:
@kovrus what was the choice on the OTel Collector Span Metrics Connector? cc @jpkrohling 
I guess there is no value in creating a `target_info` if `service.instance.id` is missing

Reply:
Please note that this drifting towards the connector and not the processor. 

In the connector, not having a `service.instance.id` will mean that `target_info` is indeed generated. However, I don't think there is any point to it. I am okay to sync with upstream behavior:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5f5770535d75a51ca07c6ea4e19906b0fa7add9c/pkg/translator/prometheusremotewrite/helper.go#L188-L205

1. If `service.name` is present, then `job` is present. If not, the label is skipped. If ONLY `service.namespace` is present and not `service.name`, then also `job` is skipped.
2. If `service.instance.id` is present then `instance` is present. If not, the label is skipped.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheusremotewrite/helper.go#L536-L550

If there are NO resource labels other than the `job` and `instance` labels, then the `target_info` is skipped entirely.


-------------

@kovrus What do you think about changing the upstream behavior? Skip `target_info` generation if `service.instance.id` is not present?